### DISCUSSION TITLE
ei_data SDK PHP is Empty

### DISCUSSION THREAD

**Discussion by thebetti** [fattureincloud/api#225] [Q&A] [answered]
Buongiorno, il campo ei_data non è compilato
e_invoice (Boolean) TRUE
ei_data (Object) stdClass
vat_kind (NULL)
original_document_type (NULL)
od_number (String, 0 characters )
od_date (NULL)
cig (String, 0 characters )
cup (String, 0 characters )
payment_method (String, 4 characters ) ***
bank_name (String, 0 characters )
bank_iban (String, 27 characters ) ******
bank_beneficiary (String, 34 characters ) ******
invoice_number (NULL)
invoice_date (NULL)

**Comment by emajo** [comment, marked as answer]
Ciao @thebetti, con solo questi dati ci viene difficile capire l'origine del problema, sarebbe utile avere il json completo della richiesta che può essere ottenuto facendo `json_encode($fattura)`.

Grazie,
Emanuele.

**Reply by thebetti** [reply]
Salve, scusate ma ho sbagliato in copia incolla.

Quando effettuo una ricerca con la libreria SDK PHP :
	$result = $apiInstance->getIssuedDocument($company_id, $document_id, $fields, $fieldset);
ed analizzo la risposta :
   $responce_ser = $result->jsonSerialize();
   $resp_ei_dataObj=ObjectSerializer::deserialize($responce_ser->data->ei_data, '\FattureInCloud\Model\IssuedDocumentEiData');

l'oggetto della ricerca ritornato è il seguente:

`ei_data (Object) stdClass :
vat_kind (NULL)
original_document_type (NULL)
od_number (String, 0 characters )
od_date (NULL)
cig (String, 0 characters )
cup (String, 0 characters )
payment_method (String, 4 characters ) ***
bank_name (String, 0 characters )
bank_iban (String, 27 characters ) ******
bank_beneficiary (String, 34 characters ) ******
invoice_number (NULL)
invoice_date (NULL)`

**Reply by emajo** [reply]
@thebetti $result è già un oggetto di classe IssuedDocument e al suo interno ha già a sua volta un oggetto di classe IssuedDocumentEiData, di conseguenza quegli step di serializzazione/deserializzazione successivi sono inutili, $resp_ei_dataObj = `$result->getData()->getEiData()` fa la stessa cosa.
Più che la risposta sarebbe più utile il json della chiamata di creazione, per vedere se è stato effettivamente popolato in creazione ei_data, quali sarebbero i campi che non stanno venendo ritornati correttamente in ei_data?

**Reply by valmoz** [reply]
Ciao @thebetti sei riuscito a risolvere?
